bson/bsoncodec/map_codec.go

```diff
@@ -26,7 +26,7 @@ type MapCodec struct {
 	EncodeKeysWithStringer bool
 }
 
-var _ ValueCodec = &MapCodec{}
+var _ valueCodec = &MapCodec{}
```

Comment:
This compile-time assertion is actually redundant because the codec is already passed into `RegisterTypeEncoder` and `RegisterTypeDecoder`, so I will remove it.